### PR TITLE
Add `BaseMultiTable` property

### LINKED ISSUE #357: Create multi table BaseProperty class
### Problem Description

_As a developer, it would be easier to make changes to the `QualityReport` if our code abstractions matched our logical and user facing abstractions. It's hard to add new properties, modify aggregation in existing ones and handle errors now because the results are being collected in a way that doesn't match the desired output._
 
After adding the `BaseSingleTableProperty` class in #354, we can use it at the multi table level as well. To do this we can create a `BaseMultiTableProperty` class and add it to a `_properties` module in the `reports/multi_table` folder.

#### Attributes

- `metrics`: A list of metrics that make up the property.
- `_properties`: A dict mapping the table names to their single table properties.

#### Abstract methods

- `get_score` - Returns a float that is the average score of all the individual metric scores computed. This method should average all the scores for each column/column pair as opposed to getting the average score for each table and averaging those.
- `get_visualization(table_name)` - Returns a `plotly.graph_objects._figure.Figure` object for the specified table.

### Additional context

- In the case of column shape metrics, the score should be computed by averaging the scores of every column in every table. This way no table scores are weighted disproportionately. For column pair metrics, the same should happen except for every column pair for each table.

### PR DESCRIPTION
**Pull request by pvk-developer**
Resolves #357

### CODE REVIEW

File: tests/unit/reports/multi_table/_properties/test_base.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    # Setup
+    base_property = BaseMultiTableProperty()
+
+    # Run and Assert
+    with pytest.raises(NotImplementedError):
+        base_property.get_score(None, None, None, None)
+
+
+def test_get_visualization():
+    """Test that the method returns the propertie's ``get_visualization``."""
```
Comment (fealho) [inline tests/unit/reports/multi_table/_properties/test_base.py:21]: typo: properties'

**Review by fealho — APPROVED** [review]

File: tests/unit/reports/multi_table/_properties/test_base.py (hunk 1 above)
Comment (amontanez24) [reply tests/unit/reports/multi_table/_properties/test_base.py:21]: property's

File: sdmetrics/reports/multi_table/_properties/base.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    and applies different base-level metrics based on the data type.
+
+    Attributes:
+        metrics (list):
+            A list of metrics that make up the property.
+        properties (dict):
+            A dict mapping the table names to their single table properties.
+    """
+
+    metrics = None
```
Comment (amontanez24) [inline sdmetrics/reports/multi_table/_properties/base.py:17]: I think we actually might not need this anymore

**Review by amontanez24 — APPROVED** [review]
Just some small comments but otherwise looks good

**Pull request merged**